core/domain/user_services.py

```diff
@@ -44,7 +49,10 @@ def __init__(
         self.last_agreed_to_terms = last_agreed_to_terms
         self.last_started_state_editor_tutorial = (  # pylint: disable=invalid-name
             last_started_state_editor_tutorial)
-        self.profile_picture_data_url = profile_picture_data_url
+        if profile_picture_data_url:
+            self.profile_picture_data_url = profile_picture_data_url
+        else:
+            self.profile_picture_data_url = self.fetch_gravatar(email)
```

Comment:
(To be clearer: the gravatar fetching should only happen once when a new user signs up. Otherwise we store a null image.)